### PR TITLE
Vector Type

### PR DESCRIPTION
**Pull request by MarkDuckworth**
Add support for reading and writing Firestore vectors.

### CODE REVIEW

File: packages/firestore/test/integration/api/database.test.ts [hunk 1]
```diff
@@ -576,6 +580,229 @@ apiDescribe('Database', persistence => {
...
+      ];
+
+      const docs = docsInOrder.reduce((obj, doc) => {
+        obj[Math.random().toString()] = doc;
+        return obj;
+      }, {} as { [i: string]: DocumentData });
+
+      return withTestCollection(persistence, docs, async randomCol => {
+        // We validate that the SDK orders the vector field the same way as the backend
+        // by comparing the sort order of vector fields from a Query.get() and
```
Comment (wu-hui) [inline packages/firestore/test/integration/api/database.test.ts:713]: This should be getFromServer?

**Review by wu-hui — APPROVED** [review]

File: packages/firestore/test/integration/api/database.test.ts (hunk 1 above)
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/database.test.ts]: Good point. Fixed.

**Review by hsubox76 — APPROVED** [review]

File: docs-devsite/firestore_.md [hunk 2]
```diff
@@ -155,6 +157,7 @@ https://github.com/firebase/firebase-js-sdk
...
 |  [Transaction](./firestore_.transaction.md#transaction_class) | A reference to a transaction.<!-- -->The <code>Transaction</code> object passed to a transaction's <code>updateFunction</code> provides the methods to read and write data within the transaction context. See [runTransaction()](./firestore_.md#runtransaction_6f03ec4)<!-- -->. |
+|  [VectorValue](./firestore_.vectorvalue.md#vectorvalue_class) | Represent a vector type in Firestore documents. Create an instance with . VectorValue |
```
Comment (egilmorez) [inline docs-devsite/firestore_.md:160]: Should these instances be "Represents" and "Creates" throughout?

File: packages/firestore/src/lite-api/user_data_reader.ts [hunk 3]
```diff
@@ -901,13 +907,46 @@ function parseScalarValue(
...
+    return parseVectorValue(value, context);
   } else {
     throw context.createError(
       `Unsupported field value: ${valueDescription(value)}`
     );
   }
 }
 
+/**
+ * Creates a new VectorValue proto value (using the internal format).
```
Comment (egilmorez) [inline packages/firestore/src/lite-api/user_data_reader.ts:920]: Is this a literal?  Backticks if so.

**Review by egilmorez — APPROVED** [review]
LG with tiny nits, thanks!

In a review this size, it would be great to have a way to know which files are not public-facing, like an exclusion list.  I'll ask Christina about that.

File: packages/firestore/src/lite-api/user_data_reader.ts (hunk 3 above)
Comment (MarkDuckworth) [reply packages/firestore/src/lite-api/user_data_reader.ts:920]: Thanks! updated.

File: docs-devsite/firestore_.md (hunk 2 above)
Comment (MarkDuckworth) [reply docs-devsite/firestore_.md:160]: Thanks. Fixed.

**Pull request merged**